audacity/audacity
# Pending track nonsense

Resolves: #5460

Clean up an old mess in TrackList, which predated the invention of ClientData.

Accomplish the same purpose non-intrusively now in TrackList.

That purpose is, to paint temporary recording tracks that are not the same as those in the project,
while the recording is not yet committed to undo history.  Some undo/redo items can be pushed or
modified during recording (as with Ctrl+B to drop a label), and those must be sequenced in undo
history entirely before the recording although they happened simultaneously with it.

QA: 
- Observe correct display updates during recording:
  - [x] recording to new tracks (Shift + R)
  - [x] append-recording (R key, when a track exists)
  - [x] Punch-and-roll recording
- Do things that push or modify undo history during recording; then observe undo-redo behavior.  The other changes are supposed to be sequenced all before the recording when you undo/redo
  - [x] changing the selection 
  - [x] Dropping a label (Ctrl+B, or Ctrl+M / MacOs Ctrl+.)




Recommended:
